표기단계 조절 에러 픽스 & 리팩토링

Eunsol0410 opened the pull request:
## 주요 사항
- 표기단계 조절에서 layer를 가지고 직접 맵스타일링 하던 부분을 setFeatureState 이용하도록 수정하였습니다. 
- 기존 상태 업데이트는 replaceStyle을 사용하고 있었는데, 이는 전체 스타일 스토어 상태를 다 바꾸는 방법이고, isChanged도 다 체크해준 완전한 상태를 전달해야해서 방법 수정하였습니다.
- setWholeStyle과 같은 처리를 하지만, 입력 값이 있는 것에 한해서만 실행될 수 있도록 replaceFeatureStyle 액션을 추가하였습니다.
- 현재 상태 + road 단계의 가시성만 변경 한 뒤 road 스토어만 업데이트 될 수 있고, isChanged도 체크될 수 있도록 하였습니다.
- 표기단계 조절 히스토리 로그 생성도 해당 Hook으로 이동하였습니다.
![image](https://user-images.githubusercontent.com/41146374/102016253-529e4600-3da3-11eb-91db-d24e983536b7.png)

Eunsol0410 committed:
[fix] VisibilityValueType 오타 수정

Eunsol0410 committed:
[fix] 기본 스타일에서 빠진 초기값 추가

Eunsol0410 committed:
[fix] 스타일 상태의 element 속성이 null인 예외처리

- 해당 속성이 없어서 null인 경우가 있어서 에러 발생하였음

Eunsol0410 committed:
[fix] 표기단계조절 Hook 에러 픽스와 리팩토링

- layer를 가져와서 map-styling 함수 부르지 않고, setFeatureState 이용하도록 수정하였음
- 기존 replaceStyle은 전체를 다 바꾸고, isChanged도 다시 체크한 완전상태를 전달해야함 그래서 방법 수정하였음
- setWholeStyle 액션을 사용하되, 해당 feature의 상태만 전달하여 하나의 스토어만 업데이트 되도록함
- 히스토리 로그 생성도 해당 Hook으로 이동하였음

Eunsol0410 committed:
[fix] 표기단계 조절 관련 히스토리 로그 형태 수정

- 어떤 유형인지도 표기해주기 위해서 feature도 함께 저장하도록 하였음
- useWholeStyle에서 repaceStyle로 로그 남기는건 표기단계 조절만 해당했어서, 여기서는 제거하였음

qkrdmstlr3 commented on src/hooks/sidebar/useSidebarDepthItem.ts:
```diff
@@ -129,42 +94,63 @@ function useSidebarDepthItem(
-        changedValue: changedDepth,
-      });
-    });
+    dispatch(setWholeStyle(changeStyleState));
```
setWholeStyle을 실행할 경우 기존에 고급설정에서 변경해놓은 다른 여러 상태들이 초기화되는 문제는 없을까요??

Eunsol0410 replied:
아! 그래서 다시 수정했습니다 ㅎㅎ 
새로운 액션 타입을 만들고 그거는 조건문을 넣어서, 없으면 상태 업데이트 하지 않도록 했습니다.

qkrdmstlr3 reviewed (commented):
github에서 변경사항은 항상 보기가 어렵네요... 받아서 복습해야겠습니다..! 수고하셨습니다!
사진에서 1 > 2 > 3 표기 대신에 소,중,대또는 상, 중, 하로 바꾸는 것은 어떨까요?

Eunsol0410 committed:
[fix] set_whole로 표기 단계 조절 관련 에러 픽스

- 전체 스타일 상태가 같이 업데이트 되어서, 동일한 처리를 replace_feature인지 체크해서 실행될 수 있도록 수정하였음
- set_whole은 빈 값이 와도 디폴트로 다 돌리지만, replace_feature는 본인의 feature가 비어있으면 기존 상태 유지하도록 함

Eunsol0410 committed:
[feat] 크리스마스 테마 추가

Eunsol0410 commented:
상/중/하 좋은 것 같습니다! 바로 수정해서 커밋할게요~
